### ISSUE TITLE
[On windows] Root of nvim-tree changed unexpectedly and buffer not listed by telescope

### ISSUE THREAD

**Issue by ljie-PI** [nvim-tree/nvim-tree.lua#2961]
### Description

With this change (https://github.com/nvim-tree/nvim-tree.lua/commit/45a93d99794fff3064141d5b3a50db98ce352697#diff-eeb21711fceb0abba1450bd1adcbecc4fd90cf8ce4d5d7a8233f886c1bebd41fR375), I found it doesn't work well on Windows for 2 cases:
1. When I open a file from nvim-tree, the root of nvim-tree window changed to the directory of that file
2. The buffer of this file can not be listed when I use telescope command `<cmd>lua require('telescope.builtin').buffers(require('telescope.themes').get_dropdown{previewer = false})<cr>`

### Neovim version

```text
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1713484068
```

### Operating system and version

Windows 11

### Windows variant

Command Prompt

### nvim-tree version

2a268f6

### Clean room replication

```lua
I'm on windows so don't have /tmp/nvt-min.lua.

Pasting my config here:

local status_ok, nvim_tree = pcall(require, "nvim-tree")
if not status_ok then
  return
end

local api_status_ok, nvim_tree_api = pcall(require, "nvim-tree.api")
if not api_status_ok then
  return
end

local function customized_on_attach(bufnr)
  local function opts(desc)
    return { desc = "nvim-tree: " .. desc, buffer = bufnr, noremap = true, silent = true, nowait = true }
  end
  -- default mappings
  nvim_tree_api.config.mappings.default_on_attach(bufnr)
  -- custom mappings
  vim.keymap.set('n', 'c', nvim_tree_api.tree.change_root_to_node, opts('CD'))
  vim.keymap.set('n', 'l', nvim_tree_api.node.open.edit, opts('Open'))
  vim.keymap.set('n', '<CR>', nvim_tree_api.node.open.edit, opts('Open'))
  vim.keymap.set('n', 'o', nvim_tree_api.node.open.edit, opts('Open'))
  vim.keymap.set('n', 'h', nvim_tree_api.node.navigate.parent_close, opts('Close Directory'))
  vim.keymap.set('n', 'v', nvim_tree_api.node.open.vertical, opts('Open: Vertical Split'))
end

local keymap = vim.api.nvim_set_keymap
keymap("n", "<leader>e", ":NvimTreeToggle<CR>", { noremap = true, silent = true })

nvim_tree.setup {
  update_focused_file = {
    enable = true,
    update_cwd = true,
  },
  actions = {
    change_dir = {
      enable = false,
    }
  },
  renderer = {
    root_folder_modifier = ":t",
    icons = {
      glyphs = {
        default = "",
        symlink = "",
        folder = {
          arrow_open = "",
          arrow_closed = "",
          default = "",
          open = "",
          empty = "",
          empty_open = "",
          symlink = "",
          symlink_open = "",
        },
        git = {
          unstaged = "",
          staged = "S",
          unmerged = "",
          renamed = "➜",
          untracked = "U",
          deleted = "",
          ignored = "◌",
        },
      },
    },
  },
  diagnostics = {
    enable = true,
    show_on_dirs = true,
    icons = {
      hint = "",
      info = "",
      warning = "",
      error = "",
    },
  },
  view = {
    width = 45,
    side = "left",
  },
  on_attach = customized_on_attach,
}
```

### Steps to reproduce

1. cd C:\Users\${username}\AppData\Local\nvim-data\site\pack\packer\start\nvim-tree.lua
2. nvim .
3. move to the file lua\nvim-tree\actions\root\change-dir.lua and open it
![image](https://github.com/user-attachments/assets/fa8bf439-3e78-4dd6-8177-c9170e5c4a27)
4. look at the nvim-tree window, the root changes to lua\nvim-tree\actions\root
![image](https://github.com/user-attachments/assets/f2bc2e5c-0ee9-4324-a33f-dda91d314430)
5. and call the list buffer function in telescope, nothing shows up
![image](https://github.com/user-attachments/assets/0b445a06-e776-4c77-b759-599af619725c)

**Comment by ljie-PI** [comment]
Added some debugging messages as follows:
![image](https://github.com/user-attachments/assets/93c3d033-5a66-4a35-abb9-6966e86ed44a)

And got these messages:
![image](https://github.com/user-attachments/assets/8f6125e9-1d03-43a9-a6cf-5a8ed58d2007)

The `vim_cmd` is under `path` and should not change root, but now it seems not because of "/" doesn't match "\"

**Comment by ljie-PI** [comment]
If I use telescope to open this file, everything goes well:
![image](https://github.com/user-attachments/assets/4837e2f9-d7bc-4ffa-a803-c618cec83976)
![image](https://github.com/user-attachments/assets/09ff74ed-4bd2-440e-b155-eaae5c90b570)

**Comment by ljie-PI** [comment]
I proposed a PR(https://github.com/nvim-tree/nvim-tree.lua/pull/2962) to revert previous change and fix the current issues (which I think are more severe)

**Comment by ljie-PI** [comment]
I found the [issue #2862 is more likely an issue of neovim instead of nvim-tree

**Comment by ljie-PI** [comment]
I think for later release, we can deal with this specific case separately.

**Comment by ljie-PI** [comment]
Now also made changes to fix #2862. It works for all cases (open file correctly, root not changed unexpectedly, telescope can list all buffers)
![image](https://github.com/user-attachments/assets/f790e500-17a1-46c3-b9e7-f2221a0ee5e4)

But telescope could not correctly open the files with both "()" and "[]" in path. May need changes in telescope or neovim itself to get that fixed.

**Comment by ljie-PI** [comment]
linke the PR: https://github.com/nvim-tree/nvim-tree.lua/pull/2962
